File: backend/Labelix-Backend/Labelix.Contracts/Models/IData.cs
namespace Labelix.Contracts.Persistence
{
    public interface IData : IIdentifiable
    {
        public int ProjectId { get; set; }
        public string Name { get; set; }
        public string Format { get; set; }
        public string Base64 { get; set; }
        public double Width { get; set; }
        public double Height { get; set; }
    }
}

File: frontend/labelix-web-app/src/app/utility/contracts/IData.ts
import {IIdentifiable} from './IIdentifiable';

export interface IData extends IIdentifiable {
  // is the base64 string
  Data: string;
  imageId: number;
  projectId: number;
  format: string;
  name: string;
  // die Höhe und die Breite des Bildes braucht man hier, um die Response vom AI Backend richtig
  // auszulesen ansonsten kann man diese werte ignorieren und bzw auf -1 stellen
  width: number;
  height: number;
}
